game-core/src/main/java/games/strategy/engine/data/changefactory/units/BombingUnitDamageChange.java

```diff
@@ -0,0 +1,71 @@
+package games.strategy.engine.data.changefactory.units;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.triplea.java.collections.IntegerMap;
+
+/**
+ * A game data change that captures the damage caused to a collection of units by a bombing attack.
+ */
+public class BombingUnitDamageChange extends Change {
+  private static final long serialVersionUID = 7093658184880237574L;
+  private final IntegerMap<String> hits;
+  private final IntegerMap<String> undoHits;
+  private final Collection<String> territoriesToNotify;
+
+  private BombingUnitDamageChange(
```

Comment:
Any thoughts to use an `@AllArgsConstructor(visibility = AccessLevel.PRIVATE)` here to avoid the manually written out constructor?

Reply:
I could.  This is just copied code so I didn't make any major changes except the non-primitive -> primitive change.

Reply:
Moving code is one thing, in this case we are re-writing and in my 2 cents treat this is as new code.